cloudinit/sources/DataSourceAzure.py

```diff
@@ -763,9 +763,9 @@ def crawl_metadata(self):
                 "Disable password retrieved from IMDS: %s",
                 imds_disable_password,
             )
-            crawled_data["metadata"][
-                "disable_password"
-            ] = imds_disable_password
+            crawled_data["cfg"]["ssh_pwauth"] = not imds_disable_password
```

Comment:
I think if we change the first `if imds_disable_password:` block to `if imds_disable_password is not None:`, then we should be able to drop the extra block:

```
        if imds_disable_password is False:
            crawled_data["cfg"]["ssh_pwauth"] = True
```

Reply:
that way we don't touch ssh_pwauth here if None, and set it correctly otherwise

Reply:
done